src/drivers/px4io/px4io.cpp

```diff
@@ -996,19 +999,14 @@ int PX4IO::io_handle_status(uint16_t status)
 	/**
 	 * Get and handle the safety status
 	 */
-	const bool safety_off = status & PX4IO_P_STATUS_FLAGS_SAFETY_OFF;
+	const bool trigger_safety_button = status & PX4IO_P_STATUS_FLAGS_SAFETY_OFF;
 
-	// publish immediately on change, otherwise at 1 Hz
-	if ((hrt_elapsed_time(&_safety.timestamp) >= 1_s)
-	    || (_safety.safety_off != safety_off)) {
-
-		_safety.safety_switch_available = true;
-		_safety.safety_off = safety_off;
-		_safety.timestamp = hrt_absolute_time();
-
-		_to_safety.publish(_safety);
+	if (_previous_trigger_safety_button != trigger_safety_button) {
```

Comment:
Is this working when forcing the safety on disarm, and then pressing the button again?

Reply:
I forgot to write in the description about it. 
"forcing the safety after disarm" is **not supported** for PX4IO after this PR. Not because of this code here but because of the state that doesn't change after the button is pressed on PX4IO firmware. This will need to be upgraded additionally.